Remove mapping types

As part of v2.0 release, `mapping types` are getting removed from OpenSearch engine. Below are the changes in the opensearch-engine.
- Removal of type from rest end-points
- Removal of include_type_name parameter from API requests.

As part of this issue, please verify if type removal change on OpenSearch engine impacts this repository. If yes, then please remove the type references/usage from this plugin. 


Top level changes are captured in gist below: 
https://gist.github.com/dreamer-89/d76eaf639171e8ab32fa7f8b9d6c93d3

For more detailed changes, please check meta issue below:
Related: [OpenSearch-engine meta issue](https://github.com/opensearch-project/opensearch/issues/1940)

Remove mapping type issues shows in #118 
```
/home/runner/work/asynchronous-search/asynchronous-search/src/main/java/org/opensearch/search/asynchronous/service/AsynchronousSearchPersistenceService.java:330: error: cannot find symbol
> Task :compileJava
        client.admin().indices().prepareCreate(ASYNC_SEARCH_RESPONSE_INDEX).addMapping(MAPPING_TYPE, mapping())
                                                                           ^
  symbol:   method addMapping(String,XContentBuilder)
  location: class CreateIndexRequestBuilder
1 error

> Task :compileJava FAILED
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -30,10 +30,10 @@ jobs:
       - name: Checkout Branch
         uses: actions/checkout@v2
       - name: Build with Gradle
-        run: ./gradlew build -Dopensearch.version=2.0.0-SNAPSHOT
+        run: ./gradlew build -Dopensearch.version=2.0.0-alpha1-SNAPSHOT
       - name: Pull and Run Docker for security tests
         run: |
-          version=2.0.0-SNAPSHOT
+          version=2.0.0-alpha1-SNAPSHOT
           plugin_version=2.0.0.0-SNAPSHOT
           pwd=`pwd`
           echo $pwd
@@ -108,7 +108,7 @@ jobs:
       - name: Checkout Branch
         uses: actions/checkout@v2
       - name: Build with Gradle
-        run: ./gradlew.bat build -D"opensearch.version=2.0.0-SNAPSHOT" -x integTest -x jacocoTestReport
+        run: ./gradlew.bat build -D"opensearch.version=2.0.0-alpha1-SNAPSHOT" -x integTest -x jacocoTestReport
         env:
           _JAVA_OPTIONS: -Xmx4096M
       - name: Create Artifact Path
@@ -136,7 +136,7 @@ jobs:
       - name: Checkout Branch
         uses: actions/checkout@v2
       - name: Build with Gradle
-        run: ./gradlew build -Dopensearch.version=2.0.0-SNAPSHOT -x integTest -x jacocoTestReport
+        run: ./gradlew build -Dopensearch.version=2.0.0-alpha1-SNAPSHOT -x integTest -x jacocoTestReport
         env:
           _JAVA_OPTIONS: -Xmx4096M
       - name: Create Artifact Path

--- a/.github/workflows/multi-node-test-workflow.yml
+++ b/.github/workflows/multi-node-test-workflow.yml
@@ -27,7 +27,7 @@ jobs:
       - name: Checkout Branch
         uses: actions/checkout@v2
       - name: Run integration tests with multi node config
-        run: ./gradlew integTest -PnumNodes=5 -Dopensearch.version=2.0.0-SNAPSHOT
+        run: ./gradlew integTest -PnumNodes=5 -Dopensearch.version=2.0.0-alpha1-SNAPSHOT
       - name: Run Backwards Compatibility Tests
         run: |
           echo "Running backwards compatibility tests ..."

--- a/build.gradle
+++ b/build.gradle
@@ -10,10 +10,14 @@ buildscript {
         distribution = 'oss-zip'
         opensearch_group = "org.opensearch"
         isSnapshot = "true" == System.getProperty("build.snapshot", "true")
-        opensearch_version = System.getProperty("opensearch.version", "2.0.0-SNAPSHOT")
-        // 1.0.0 -> 1.0.0.0, and 1.0.0-SNAPSHOT -> 1.0.0.0-SNAPSHOT
+        opensearch_version = System.getProperty("opensearch.version", "2.0.0-alpha1-SNAPSHOT")
         opendistro_plugin_version = System.getProperty("bwc.version", "1.13.0.1")
-        opensearch_build = opensearch_version.replaceAll(/(\.\d)([^\d]*)$/, '$1.0$2')
+        // 2.0.0-alpha1-SNAPSHOT -> 2.0.0.0-alpha1-SNAPSHOT
+        version_tokens = opensearch_version.tokenize('-')
+        opensearch_build = version_tokens[0] + '.0'
+        for (int j = 1; j < version_tokens.size(); j++) {
+            opensearch_build = opensearch_build + '-' + version_tokens[j]
+        }
         common_utils_version = System.getProperty("common_utils.version", opensearch_build)
         buildVersionQualifier = System.getProperty("build.version_qualifier")
     }
@@ -78,13 +82,7 @@ targetCompatibility = JavaVersion.VERSION_11
 
 allprojects {
     group 'org.opensearch'
-    version = opensearch_version.tokenize('-')[0] + '.0'
-    if (buildVersionQualifier) {
-        version += "-${buildVersionQualifier}"
-    }
-    if (isSnapshot) {
-        version += "-SNAPSHOT"
-    }
+    version = opensearch_build
 }
 
 dependencies {

--- a/build.gradle
+++ b/build.gradle
@@ -12,14 +12,17 @@ buildscript {
         isSnapshot = "true" == System.getProperty("build.snapshot", "true")
         opensearch_version = System.getProperty("opensearch.version", "2.0.0-alpha1-SNAPSHOT")
         opendistro_plugin_version = System.getProperty("bwc.version", "1.13.0.1")
+        buildVersionQualifier = System.getProperty("build.version_qualifier")
         // 2.0.0-alpha1-SNAPSHOT -> 2.0.0.0-alpha1-SNAPSHOT
         version_tokens = opensearch_version.tokenize('-')
         opensearch_build = version_tokens[0] + '.0'
-        for (int j = 1; j < version_tokens.size(); j++) {
-            opensearch_build = opensearch_build + '-' + version_tokens[j]
+        if (buildVersionQualifier) {
+            opensearch_build += "-${buildVersionQualifier}"
+        }
+        if (isSnapshot) {
+            opensearch_build += "-SNAPSHOT"
         }
         common_utils_version = System.getProperty("common_utils.version", opensearch_build)
-        buildVersionQualifier = System.getProperty("build.version_qualifier")
     }
 
     repositories {

--- a/src/main/java/org/opensearch/search/asynchronous/service/AsynchronousSearchPersistenceService.java
+++ b/src/main/java/org/opensearch/search/asynchronous/service/AsynchronousSearchPersistenceService.java
@@ -392,7 +392,7 @@ public class AsynchronousSearchPersistenceService {
     private XContentBuilder mapping() {
         try {
             XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
-            builder
+            builder.startObject()
                     .startObject("properties")
                     .startObject(START_TIME_MILLIS)
                     .field("type", "date")
@@ -407,6 +407,7 @@ public class AsynchronousSearchPersistenceService {
                     .endObject()
                     .startObject(ERROR)
                     .field("type", "binary")
+                    .endObject()
                     .endObject()
                     .endObject();
             return builder;

--- a/src/main/java/org/opensearch/search/asynchronous/service/AsynchronousSearchPersistenceService.java
+++ b/src/main/java/org/opensearch/search/asynchronous/service/AsynchronousSearchPersistenceService.java
@@ -67,7 +67,6 @@ public class AsynchronousSearchPersistenceService {
 
     private static final Logger logger = LogManager.getLogger(AsynchronousSearchPersistenceService.class);
     public static final String ASYNC_SEARCH_RESPONSE_INDEX = ".opendistro-asynchronous-search-response";
-    private static final String MAPPING_TYPE = "_doc";
     /**
      * The backoff policy to use when saving a asynchronous search response fails. The total wait
      * time is 600000 milliseconds, ten minutes.
@@ -327,7 +326,7 @@ public class AsynchronousSearchPersistenceService {
 
     private void createIndexAndDoStoreResult(String id, AsynchronousSearchPersistenceModel persistenceModel,
                                              ActionListener<IndexResponse> listener) {
-        client.admin().indices().prepareCreate(ASYNC_SEARCH_RESPONSE_INDEX).addMapping(MAPPING_TYPE, mapping())
+        client.admin().indices().prepareCreate(ASYNC_SEARCH_RESPONSE_INDEX).setMapping(mapping())
                 .setSettings(indexSettings()).execute(ActionListener.wrap(createIndexResponse -> doStoreResult(id, persistenceModel,
                 listener), exception -> {
             if (ExceptionsHelper.unwrapCause(exception) instanceof ResourceAlreadyExistsException) {
@@ -393,7 +392,7 @@ public class AsynchronousSearchPersistenceService {
     private XContentBuilder mapping() {
         try {
             XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
-            builder.startObject()
+            builder
                     .startObject("properties")
                     .startObject(START_TIME_MILLIS)
                     .field("type", "date")
@@ -408,7 +407,6 @@ public class AsynchronousSearchPersistenceService {
                     .endObject()
                     .startObject(ERROR)
                     .field("type", "binary")
-                    .endObject()
                     .endObject()
                     .endObject();
             return builder;

--- a/src/test/java/org/opensearch/search/asynchronous/integTests/AsynchronousSearchQueryIT.java
+++ b/src/test/java/org/opensearch/search/asynchronous/integTests/AsynchronousSearchQueryIT.java
@@ -11,12 +11,12 @@
 
 package org.opensearch.search.asynchronous.integTests;
 
+import org.apache.lucene.tests.analysis.MockTokenizer;
 import org.opensearch.search.asynchronous.plugin.AsynchronousSearchPlugin;
 import org.opensearch.search.asynchronous.request.GetAsynchronousSearchRequest;
 import org.opensearch.search.asynchronous.request.SubmitAsynchronousSearchRequest;
 import org.opensearch.search.asynchronous.response.AsynchronousSearchResponse;
 import org.opensearch.search.asynchronous.utils.TestClientUtils;
-import org.apache.lucene.analysis.MockTokenizer;
 import org.apache.lucene.analysis.pattern.PatternReplaceCharFilter;
 import org.opensearch.action.admin.indices.refresh.RefreshRequest;
 import org.opensearch.action.search.SearchRequest;
@@ -120,14 +120,14 @@ public class AsynchronousSearchQueryIT extends OpenSearchIntegTestCase {
         // statement
         assertAcked(prepareCreate("test")
                 .setSettings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, NUM_SHARDS).put(SETTING_NUMBER_OF_REPLICAS, 0))
-                .addMapping(
+                .setMapping(
                         "book", "author", "type=keyword", "name", "type=text", "genre",
                         "type=keyword", "price", "type=float"));
         createIndex("idx_unmapped");
         // idx_unmapped_author is same as main index but missing author field
         assertAcked(prepareCreate("idx_unmapped_author")
                 .setSettings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, NUM_SHARDS).put(SETTING_NUMBER_OF_REPLICAS, 0))
-                .addMapping("book", "name", "type=text", "genre", "type=keyword", "price", "type=float"));
+                .setMapping("book", "name", "type=text", "genre", "type=keyword", "price", "type=float"));
 
         ensureGreen();
         String data[] = {
@@ -184,7 +184,7 @@ public class AsynchronousSearchQueryIT extends OpenSearchIntegTestCase {
 
     public void testIpRangeQuery() throws InterruptedException {
         assertAcked(prepareCreate("idx")
-                .addMapping("type", "ip", "type=ip", "ips", "type=ip"));
+                .setMapping("type", "ip", "type=ip", "ips", "type=ip"));
         waitForRelocation(ClusterHealthStatus.GREEN);
 
         indexRandom(true,
@@ -245,7 +245,7 @@ public class AsynchronousSearchQueryIT extends OpenSearchIntegTestCase {
                 .endObject().endObject();
         mappings.endObject();
         assertAcked(prepareCreate("test1")
-                .addMapping("type", mappings));
+                .setMapping(mappings));
         client().prepareIndex("test1").setId("1")
                 .setSource(jsonBuilder().startObject().field("text", "foo").endObject())
                 .get();

--- a/src/test/java/org/opensearch/search/asynchronous/integTests/AsynchronousSearchQueryIT.java
+++ b/src/test/java/org/opensearch/search/asynchronous/integTests/AsynchronousSearchQueryIT.java
@@ -121,13 +121,12 @@ public class AsynchronousSearchQueryIT extends OpenSearchIntegTestCase {
         assertAcked(prepareCreate("test")
                 .setSettings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, NUM_SHARDS).put(SETTING_NUMBER_OF_REPLICAS, 0))
                 .setMapping(
-                        "book", "author", "type=keyword", "name", "type=text", "genre",
-                        "type=keyword", "price", "type=float"));
+                        "author", "type=keyword", "name", "type=text", "genre", "type=keyword", "price", "type=float"));
         createIndex("idx_unmapped");
         // idx_unmapped_author is same as main index but missing author field
         assertAcked(prepareCreate("idx_unmapped_author")
                 .setSettings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, NUM_SHARDS).put(SETTING_NUMBER_OF_REPLICAS, 0))
-                .setMapping("book", "name", "type=text", "genre", "type=keyword", "price", "type=float"));
+                .setMapping("name", "type=text", "genre", "type=keyword", "price", "type=float"));
 
         ensureGreen();
         String data[] = {
@@ -184,7 +183,7 @@ public class AsynchronousSearchQueryIT extends OpenSearchIntegTestCase {
 
     public void testIpRangeQuery() throws InterruptedException {
         assertAcked(prepareCreate("idx")
-                .setMapping("type", "ip", "type=ip", "ips", "type=ip"));
+                .setMapping("ip", "type=ip", "ips", "type=ip"));
         waitForRelocation(ClusterHealthStatus.GREEN);
 
         indexRandom(true,
@@ -235,15 +234,13 @@ public class AsynchronousSearchQueryIT extends OpenSearchIntegTestCase {
 
     public void testHighlighterQuery() throws IOException, InterruptedException {
         XContentBuilder mappings = jsonBuilder();
-        mappings.startObject();
-        mappings.startObject("type")
+        mappings.startObject()
                 .startObject("properties")
                 .startObject("text")
                 .field("type", "keyword")
                 .field("store", true)
                 .endObject()
                 .endObject().endObject();
-        mappings.endObject();
         assertAcked(prepareCreate("test1")
                 .setMapping(mappings));
         client().prepareIndex("test1").setId("1")

--- a/src/test/java/org/opensearch/search/asynchronous/listener/AsynchronousSearchPartialResponseIT.java
+++ b/src/test/java/org/opensearch/search/asynchronous/listener/AsynchronousSearchPartialResponseIT.java
@@ -63,7 +63,7 @@ public class AsynchronousSearchPartialResponseIT extends OpenSearchIntegTestCase
 
     protected void createIdx(String keyFieldMapping) {
         assertAcked(prepareCreate("idx")
-                .setMapping("type", "key", keyFieldMapping));
+                .setMapping("key", keyFieldMapping));
     }
 
     protected void indexData() throws Exception {

--- a/src/test/java/org/opensearch/search/asynchronous/listener/AsynchronousSearchPartialResponseIT.java
+++ b/src/test/java/org/opensearch/search/asynchronous/listener/AsynchronousSearchPartialResponseIT.java
@@ -63,7 +63,7 @@ public class AsynchronousSearchPartialResponseIT extends OpenSearchIntegTestCase
 
     protected void createIdx(String keyFieldMapping) {
         assertAcked(prepareCreate("idx")
-                .addMapping("type", "key", keyFieldMapping));
+                .setMapping("type", "key", keyFieldMapping));
     }
 
     protected void indexData() throws Exception {

--- a/src/test/java/org/opensearch/search/asynchronous/management/AsynchronousSearchManagementServiceIT.java
+++ b/src/test/java/org/opensearch/search/asynchronous/management/AsynchronousSearchManagementServiceIT.java
@@ -124,7 +124,7 @@ public class AsynchronousSearchManagementServiceIT extends AsynchronousSearchInt
     public void testDeletesExpiredAsynchronousSearchResponseFromPersistedStore() throws Exception {
         String idx = "idx";
         assertAcked(prepareCreate(idx)
-                .addMapping("type", "ip", "type=ip", "ips", "type=ip"));
+                .setMapping("type", "ip", "type=ip", "ips", "type=ip"));
         waitForRelocation(ClusterHealthStatus.GREEN);
         indexRandom(true,
                 client().prepareIndex(idx).setId("1").setSource(

--- a/src/test/java/org/opensearch/search/asynchronous/management/AsynchronousSearchManagementServiceIT.java
+++ b/src/test/java/org/opensearch/search/asynchronous/management/AsynchronousSearchManagementServiceIT.java
@@ -124,7 +124,7 @@ public class AsynchronousSearchManagementServiceIT extends AsynchronousSearchInt
     public void testDeletesExpiredAsynchronousSearchResponseFromPersistedStore() throws Exception {
         String idx = "idx";
         assertAcked(prepareCreate(idx)
-                .setMapping("type", "ip", "type=ip", "ips", "type=ip"));
+                .setMapping("ip", "type=ip", "ips", "type=ip"));
         waitForRelocation(ClusterHealthStatus.GREEN);
         indexRandom(true,
                 client().prepareIndex(idx).setId("1").setSource(

--- a/src/test/java/org/opensearch/search/asynchronous/request/AsynchronousSearchRequestRoutingIT.java
+++ b/src/test/java/org/opensearch/search/asynchronous/request/AsynchronousSearchRequestRoutingIT.java
@@ -48,7 +48,7 @@ public class AsynchronousSearchRequestRoutingIT extends AsynchronousSearchIntegT
     public void testRequestForwardingToCoordinatorNodeForPersistedAsynchronousSearch() throws Exception {
         String idx = "idx";
         assertAcked(prepareCreate(idx)
-                .addMapping("type", "ip", "type=ip", "ips", "type=ip"));
+                .setMapping("type", "ip", "type=ip", "ips", "type=ip"));
         waitForRelocation(ClusterHealthStatus.GREEN);
         indexRandom(true,
                 client().prepareIndex(idx).setId("1").setSource(
@@ -98,7 +98,7 @@ public class AsynchronousSearchRequestRoutingIT extends AsynchronousSearchIntegT
         List<ScriptedBlockPlugin> plugins = initBlockFactory();
         String index = "idx";
         assertAcked(prepareCreate(index)
-                .addMapping("type", "ip", "type=ip", "ips", "type=ip"));
+                .setMapping("type", "ip", "type=ip", "ips", "type=ip"));
         waitForRelocation(ClusterHealthStatus.GREEN);
 
         indexRandom(true,

--- a/src/test/java/org/opensearch/search/asynchronous/request/AsynchronousSearchRequestRoutingIT.java
+++ b/src/test/java/org/opensearch/search/asynchronous/request/AsynchronousSearchRequestRoutingIT.java
@@ -48,7 +48,7 @@ public class AsynchronousSearchRequestRoutingIT extends AsynchronousSearchIntegT
     public void testRequestForwardingToCoordinatorNodeForPersistedAsynchronousSearch() throws Exception {
         String idx = "idx";
         assertAcked(prepareCreate(idx)
-                .setMapping("type", "ip", "type=ip", "ips", "type=ip"));
+                .setMapping("ip", "type=ip", "ips", "type=ip"));
         waitForRelocation(ClusterHealthStatus.GREEN);
         indexRandom(true,
                 client().prepareIndex(idx).setId("1").setSource(
@@ -98,7 +98,7 @@ public class AsynchronousSearchRequestRoutingIT extends AsynchronousSearchIntegT
         List<ScriptedBlockPlugin> plugins = initBlockFactory();
         String index = "idx";
         assertAcked(prepareCreate(index)
-                .setMapping("type", "ip", "type=ip", "ips", "type=ip"));
+                .setMapping("ip", "type=ip", "ips", "type=ip"));
         waitForRelocation(ClusterHealthStatus.GREEN);
 
         indexRandom(true,

